packages/aws-lambda/src/triggers.js

```diff
@@ -69,9 +73,22 @@ function isApiGatewayProxyTrigger(event) {
   // NOTE: Gateway Rest API with lambda proxy -> always format 1.0
   // NOTE: Gateway HTTP API with lambda proxy -> default is 2.0, but can be configured on creation
   // NOTE: Gateway REST/HTTP API without lambda proxy -> format cannot be interpreted
+  // NOTE: Function URL also support version 2 and contains rawPath, Gateway HTTP API does not follow the format
+  // https://<url-id>.lambda-url.<region>.on.aws
   return (
     (event.resource != null && event.path != null && event.httpMethod != null) ||
-    (event.rawPath && event.version === '2.0')
+    (event.rawPath && event.version === '2.0' &&
+      !(event.requestContext.domainName && event.requestContext.domainName.includes('lambda-url')))
+  );
+}
+
+function isFunctionURLTrigger(event) {
+  // NOTE: Function URL -> Lambda sets routeKey to $default as a placeholder.
+  // NOTE: Function URL -> Currently support version 2.0.
+  // NOTE: Function URL -> Domain name follows the format https://<url-id>.lambda-url.<region>.on.aws
+  return (
+    event.routeKey === '$default' && event.version === '2.0' &&
+    (event.requestContext.domainName && event.requestContext.domainName.includes('lambda-url'))
```

Comment:
_suggestion:_ We might be a bit more specific here than `domainName.includes('lambda-url')`. In theory, other URLs might contain "lambda-url" as a substring. Maybe a regex match against the documented pattern `https://<url-id>.lambda-url.<region>.on.aws`? (And if we change it here, the same change would then be needed in line 81.)

Reply:
added the regex pattern